flytekit/core/annotation.py

```diff
@@ -0,0 +1,27 @@
+from typing import Any, Dict
+
+
+class FlyteAnnotation:
+    """A core object to add arbitrary annotations to flyte types.
+
+    This metadata is ingested as a python dictionary and will be serialized
+    into fields on the flyteidl type literals. This data is not accessible at
+    runtime but rather can be retrieved from flyteadmin for custom presentation
+    of typed parameters.
+
+    For a task definition:
+
+    .. code-block:: python
+
+        @task
+        def x(a: typing.Annotated[int, FlyteAnnotation({"foo": {"bar": 1}})]):
+            return
+
+    """
+
+    def __init__(self, data: Dict[str, Any]):
+        self._data = data
+
+    @property
+    def data(self):
```

Comment:
should we call this `annotations` to match the IDL?  Or leave it `data`?

Reply:
i named it `data` to reinforce that this is the client-side, python object and so it wouldn't get confused with the idl structure and the python model that represents it (eg. `FlyteAnnotation` vs. `TypeAnnotation`)